app/src/main/java/de/grobox/transportr/trips/detail/TripDetailFragment.kt

```diff
@@ -0,0 +1,178 @@
+        list.layoutManager = LinearLayoutManager(context)
+        bottomBar.setOnClickListener { _ -> onBottomBarClick() }
+
+        viewModel = ViewModelProviders.of(activity!!, viewModelFactory).get(TripDetailViewModel::class.java)
+        viewModel.getTrip().observe(this, Observer<Trip> { this.onTripChanged(it) })
+        viewModel.sheetState.observe(this, Observer<SheetState> { this.onSheetStateChanged(it) })
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        val toolbarMenu = toolbar.menu
+        inflater.inflate(R.menu.trip_details, toolbarMenu)
+        viewModel.tripReloadError.observe(this, Observer<String> { this.onTripReloadError(it) })
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onMenuItemClick(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_reload -> {
+                item.setActionView(R.layout.actionbar_progress_actionview)
+                viewModel.reloadTrip()
+                return true
+            }
+            R.id.action_share -> {
+                share(context, viewModel.getTrip().value)
+                return true
+            }
+            R.id.action_calendar -> {
+                intoCalendar(context, viewModel.getTrip().value)
+                return true
+            }
+            else -> return false
+        }
+    }
+
+    private fun onTripChanged(trip: Trip?) {
+        if (trip == null) return
+
+        val reloadMenuItem = toolbar.menu.findItem(R.id.action_reload)
+        if (reloadMenuItem != null) reloadMenuItem.actionView = null
+
+        val network = viewModel.transportNetwork.value
+        val showLineName = network != null && network.hasGoodLineNames()
+        val adapter = LegAdapter(trip.legs, viewModel, showLineName)
+        list.adapter = adapter
+
+        fromTime.text = getTime(context, trip.firstDepartureTime)
+        setRelativeDepartureTime(fromTimeRel, trip.firstDepartureTime)
+        from.text = trip.from.uniqueShortName()
+        toTime.text = getTime(context, trip.lastArrivalTime)
+        to.text = trip.to.uniqueShortName()
+        duration.text = getDuration(trip.duration)
+        durationTop.text = getString(R.string.total_time, getDuration(trip.duration))
+        price.visibility = if (trip.hasFare()) VISIBLE else GONE
+        price.text = trip.getStandardFare()
+        priceTop.visibility = if (trip.hasFare()) VISIBLE else GONE
+        priceTop.text = trip.getStandardFare()
+    }
+
+    private fun onToolbarClose() {
+        viewModel.sheetState.value = BOTTOM
+    }
+
+    private fun onBottomBarClick() {
+        viewModel.sheetState.value = MIDDLE
+    }
+
+    private fun onSheetStateChanged(sheetState: SheetState?) {
+        when (sheetState) {
+            null -> return
+            BOTTOM -> {
+                toolbar.visibility = GONE
+                topBar.visibility = GONE
+                bottomBar.visibility = VISIBLE
```

Comment:
These visibility changes appear a bit rough in the UI. Should be easy to use a object animator instead here to make things more smooth. But we can do this in a separate MR.